src/util/address.rs

```diff
@@ -316,7 +337,11 @@ impl Address {
             Payload::WitnessProgram {
                 version: ver,
                 program: ref prog,
-            } => script::Builder::new().push_int(ver.to_u8() as i64).push_slice(&prog),
+            } => {
+                assert!(ver.to_u8() <= 16);
+                let verop = opcodes::all::OP_PUSHBYTES_0.into_u8() + ver.to_u8();
+                script::Builder::new().push_opcode(verop.into()).push_slice(&prog)
```

Comment:
I think the old logic was much clearer, though +1 to adding the assertion.

Reply:
Actually as near as I can tell the new logic is not only unclear, but wrong.

Reply:
I continue to think we should use `push_int` instead of this manual opcode-munging business.

Reply:
Can you rebase the fix to this? It's hard to re-review when there is a commit introducing the original bug followed by a broken bugfix, followed by(?) another bugfix.